### PR TITLE
feat: Add migration to merge course pages into courses

### PR DESCRIPTION
**Pull request by kulla**

### CODE REVIEW

File: src/20240506145600-merge-course-pages-into-courses.ts [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+              state: {
+                plugin: 'splishContent',
+                state: content,
+              },
+              id: uuidv4(),
+            },
+          ],
+          id: uuidv4(),
+        }
+      }
```
Comment (kulla) [inline src/20240506145600-merge-course-pages-into-courses.ts:207]: @elbotho I actually have found old splish content in one revision => Is this a good way to store it (I assume it is in an unused revision / course-page)

File: src/20240506145600-merge-course-pages-into-courses.ts [hunk 2]
```diff
@@ -0,0 +1,686 @@
...
+
+      if (RowPluginDecoder.is(content)) {
+        content = {
+          plugin: 'coursePage',
+          state: { content },
+        }
+      }
```
Comment (kulla) [inline src/20240506145600-merge-course-pages-into-courses.ts:214]: @elbotho That's a difference to the issue #338 => course pages are in their own plugin, okay or should they be part of the state of the course?

**Review by kulla — COMMENTED** [review]
@AndreasHuber @hugotiburtino This PR is ready to be reviewed already

File: src/20240506145600-merge-course-pages-into-courses.ts (hunk 1 above)
Comment (elbotho) [reply src/20240506145600-merge-course-pages-into-courses.ts:207]: Oh wow…

> I assume it is in an unused revision / course-page

If it's actually not in an accepted revision this is great, thanks.

**Review by hugotiburtino — APPROVED** [review]
There are some bugs (see beta testing doc) that may be related to this migration. Please take a look and decide if any changes here are necessary.

**Comment by elbotho** [comment]
@kulla 
Last time (for GroupedExercises) we had to fix the subscriptions afterwards. Do we have to do that here too?

https://github.com/serlo/db-migrations/pull/308/files#diff-5ff6abef2864f615fb1bf2d8e6a04c2b21d04099a71dceb4d5738ebbff5ae25b

**Pull request merged**